src/renderer/services/BeaconAPIClient.ts

```diff
@@ -0,0 +1,136 @@
+import {
+    uint64,
+    BLSPubkey,
+    Epoch,
+    ValidatorDuty,
+    Slot,
+    BeaconBlock,
+    uint8,
+    Shard,
+    IndexedAttestation
+} from '@chainsafe/eth2.0-types';
+import {
+    FETCH_NODE_VERSION,
+    FETCH_GENESIS_TIME,
+    POLL_NODE_SYNCING,
+    FETCH_FORK_INFORMATION,
+    FETCH_VALIDATOR_DUTIES,
+    FETCH_VALIDATOR_BLOCK,
+    PUBLISH_SIGNED_BLOCK,
+    PRODUCE_ATTESTATION,
+    PUBLISH_SIGNED_ATTESTATION
+} from '../constants/apiUrls';
+import { Syncing, ForkInformation, IBeaconAPIClient, IBeaconApiClientOptions } from './interface';
+import { HttpClient } from './httpClient';
+import { EmptyUrlError } from './errors/EmptyUrlError';
+
+export class BeaconAPIClient implements IBeaconAPIClient {
+    options: IBeaconApiClientOptions;
+    httpClient: HttpClient;
+
+    constructor(options: IBeaconApiClientOptions) {
+        if (!options.urlPrefix) {
+            throw new EmptyUrlError();
+        }
+        this.options = options;
+        this.httpClient = new HttpClient(options.urlPrefix);
+    }
+
+    async fetchNodeVersion(): Promise<string> {
+        // TODO handle error catch
+        try {
+            const result = await this.httpClient.get<string>(FETCH_NODE_VERSION);
+            return result;
+        } catch (error) {
+            return error;
```

Comment:
Just noticed this. I would remove error and let whoever is using library to catch them. You are returning error so you would have to change return type of method to be `Promise<string|Error>`.

Reply:
Will refactor this in an hour.